From 04a35b2c57b84e098c9cad75d36675a958e9075e Mon Sep 17 00:00:00 2001
From: Sam James <sam@gentoo.org>
Date: Mon, 26 Sep 2022 20:54:16 +0100
Subject: [PATCH 2/3] Fix localename type

Needs to be a string.

Fixes error:
```
/var/tmp/portage/dev-util/schroot-1.6.10_p7/work/schroot-1.6.10/sbuild/sbuild-basic-keyfile.tcc:217:18: error: no viable overloaded '='
      localename = std::locale::classic();
      ~~~~~~~~~~ ^ ~~~~~~~~~~~~~~~~~~~~~~
/usr/lib/gcc/x86_64-pc-linux-gnu/10.2.0/include/g++-v10/bits/basic_string.h:665:7: note: candidate function not viable: no known conversion from 'const std::locale' to 'const std::__cxx11::basic_string<char>' for 1st argument
      operator=(const basic_string& __str)
      ^
[...]
```

Bug: https://bugs.gentoo.org/739034
Signed-off-by: Sam James <sam@gentoo.org>
Origin: Gentoo
Upstream-Status: Unknown
Signed-off-by: Ismael Luceno <ismael@sourcemage.org>
---
 sbuild/sbuild-basic-keyfile.tcc |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/sbuild/sbuild-basic-keyfile.tcc
+++ b/sbuild/sbuild-basic-keyfile.tcc
@@ -214,7 +214,7 @@ sbuild::basic_keyfile<K, P>::get_locale_string (group_name_type const& group,
     }
   catch (std::runtime_error const& e) // Invalid locale
     {
-      localename = std::locale::classic();
+      localename = std::locale::classic().name();
     }
   std::string::size_type pos;
   bool status = false;
-- 
Fixed up by sm-checkpatch 0.1

